kubernetes/structures_rbac.go

```diff
@@ -42,13 +42,23 @@ func expandRBACSubjects(in []interface{}) []api.Subject {
 			subject.APIGroup = v.(string)
 		}
 		if v, ok := m["kind"].(string); ok {
-			subject.Kind = string(v)
+			subject.Kind = v
 		}
 		if v, ok := m["name"]; ok {
 			subject.Name = v.(string)
 		}
-		if v, ok := m["namespace"]; ok {
-			subject.Namespace = v.(string)
+
+		// Handle namespace logic
+		if subject.Kind == "Group" || subject.Kind == "User" {
```

Comment:
seems like the logic can be simplified a bit more where the condition only handles the Kind that use the namespace. See if you can simplify this a bit more. Same goes for the flattener.